### PR TITLE
163518693 User Follow Each Other

### PR DESCRIPTION
**Pull request by abayo-luc**
#### What does this PR do?
Enable user to follow auther
#### Description of Task to be completed?
```POST /api/v1/profiles/:username/follow```
```DELETE /api/v1/profiles/:username/follow```
```GET /api/v1/profiles/:username/followers```
```GET /api/v1/profiles/:username/followings```
#### How should this be manually tested?
-Clone the repo and checkout `ft-user-follow-each-other-163518693` branch
- Using any `API client tool(usually POSTMAN)`,
- Create the user and login to get the `access-token`
- Make request to the above end point with existing ```username```
- Make sure to pass the token under `request headers`. It should be {`Authorization`: `Bearer <access-token>`} KEY, VALUE PAIR
#### Any background context you want to provide?
before running anything, ensure packages are updated ```yarn``` and no pending migration ```sequelize db:migrate```
#### What are the relevant pivotal tracker stories?
163518693

### CODE REVIEW

**Review by YvesIraguha — APPROVED** [review]

File: controllers/followerController.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
+import models from "../models";
+
+const { User, Follower } = models;
+/** Define the class for creating the profile
```
Comment (puritybirir) [inline controllers/followerController.js]: Use JSDoc better. Follow standard set by Musigwa

File: controllers/profileController.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
+// import models from "../models/index";
+import models from "../models";
+import validation from "../middlewares/validators/updateProfileValidator";
+const { User, Follower } = models;
+console.log(User);
```
Comment (puritybirir) [inline controllers/profileController.js]: Remove console.log when making pull request

File: sample.env [hunk 3]
```diff
@@ -6,5 +6,16 @@ TEST_DB_NAME ='test_database_name';
...
 BCRYPT_HASH_ROUNDS = 'number, better do greater than 10'
 SENDGRID_API_KEY = '********************'
+<<<<<<< HEAD
+SERVER_HOST = '***yourhost***/ap/v1'
 DEVELOPER_EMAIL = 'sendgrid_email'
-SERVER_HOST = '***yourhost***/ap/v1'
\ No newline at end of file
+SERVER_HOST = '***yourhost***/ap/v1'
+=======
+<<<<<<< HEAD
```
Comment (puritybirir) [inline sample.env]: Why are there merge conflicts in this file?

File: tests/users/follower.test.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+      truncate: true,
+      cascade: true
+    });
+    await models.User.destroy({
+      where: {},
+      truncate: true,
+      cascade: true
+    });
+  });
+
```
Comment (puritybirir) [inline tests/users/follower.test.js:37]: Use consistent spacing

**Review by puritybirir — CHANGES_REQUESTED** [review]

**Comment by abayo-luc** [comment]
Duplicated

**Pull request closed without merging**